addon/-legacy-private/system/store.js

```diff
@@ -220,9 +220,48 @@ Store = Service.extend({
     this._serializerCache = Object.create(null);
 
     if (DEBUG) {
-      this.__asyncRequestCount = 0;
+      if (this._shouldTrackAsyncRequests === undefined) {
+        this._shouldTrackAsyncRequests = true;
+      }
+      if (this._generateStackTracesForTrackedRequests === undefined) {
+        this._generateStackTracesForTrackedRequests = false;
+      }
+
+      this._trackedAsyncRequests = [];
+      this._trackAsyncRequestStart = label => {
+        let trace =
+          'set `store._generateStackTracesForTrackedRequests = true;` to get a detailed trace for where this request originated';
+
+        if (this._generateStackTracesForTrackedRequests) {
+          try {
+            throw new Error(`EmberData TrackedRequest: ${label}`);
+          } catch (e) {
+            trace = e;
+          }
+        }
+
+        let token = Object.freeze({
+          label,
+          trace,
+        });
+
+        this._trackedAsyncRequests.push(token);
+        return token;
+      };
+      this._trackAsyncRequestEnd = token => {
+        let index = this._trackedAsyncRequests.indexOf(token);
+
+        if (index !== -1) {
+          this._trackedAsyncRequests.splice(index, 1);
+        }
```

Comment:
If their is no index, something seems wrong. We should explore/consider the impact of throwing.